src/Symfony/Bundle/FrameworkBundle/Secret/SodiumEncoder.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secret;
+
+use Symfony\Bundle\FrameworkBundle\Exception\EncryptionKeyNotFoundException;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class SodiumEncoder implements EncoderInterface
+{
+    private $encryptionKey;
+    private $encryptionKeyOrPath;
+
+    public function __construct(string $encryptionKeyOrPath)
+    {
+        if (!\function_exists('\sodium_crypto_stream_xor')) {
+            throw new \RuntimeException('The "sodium" PHP extension is not loaded.');
+        }
+
+        $this->encryptionKeyOrPath = $encryptionKeyOrPath;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generateKeys(bool $override = false): array
+    {
+        $path = $this->getKeyPath();
+        if (!$override && is_file($path)) {
+            throw new \LogicException(sprintf('A key already exist in "%s".', $path));
+        }
+
+        $this->encryptionKey = null;
+
+        $encryptionKey = \sodium_crypto_stream_keygen();
+        (new Filesystem())->dumpFile($path, $encryptionKey);
+        \sodium_memzero($encryptionKey);
+
+        return [$path];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function encrypt(string $message): string
+    {
+        $nonce = random_bytes(\SODIUM_CRYPTO_STREAM_NONCEBYTES);
+
+        $key = $this->getKey();
+        $encryptedText = \sodium_crypto_stream_xor($message, $nonce, $key);
+        \sodium_memzero($message);
+        \sodium_memzero($key);
+
+        return $this->encode($nonce, $encryptedText);
+    }
+
+    private function getKeyPath(): ?string
+    {
+        if (null === @\is_file($this->encryptionKeyOrPath)) {
+            throw new \InvalidArgumentException('The path to the encryption key is not valid. Did you configured the framework with the content of key instead of its location?');
```

Comment:
Did you configure the framework with the content of the key

Reply:
removed as we don't handle such case anymore. cf https://github.com/symfony/symfony/pull/31101#discussion_r275243236